modules/core/src/main/java/org/apache/ignite/internal/mem/unsafe/UnsafeMemoryProvider.java

```diff
@@ -97,10 +103,44 @@ public UnsafeMemoryProvider(IgniteLogger log) {
 
         long chunkSize = sizes[regions.size()];
 
+        boolean advice = IgniteSystemProperties.getBoolean(IGNITE_OFF_HEAP_MEM_ADVICE, false);
+
         long ptr;
 
+        PointerByReference refToPtr = new PointerByReference();
+
         try {
-            ptr = GridUnsafe.allocateMemory(chunkSize);
+            int pageSize = GridUnsafe.getpagesize();
+
+            if (pageSize < 0)
+                pageSize = GridUnsafe.pageSize();
+
+            if (pageSize < 0){
+                U.warn(log,"Can not get native page size, use 4k as default.");
+
+                pageSize = 4096;
+            }
+
+            int ptrRes = GridUnsafe.posix_memalign(refToPtr, new NativeLong(pageSize), new NativeLong(chunkSize));
```

Comment:
Availability of this call should be checked. It is not available e.g. on Windows